core/src/test/kotlin/org/opensearch/alerting/core/model/LocalUriInputTests.kt

```diff
@@ -0,0 +1,527 @@
+        pathParams = "index1,index2,index3,index4,index5"
+        val localUriInput = LocalUriInput(scheme, host, port, path, pathParams, queryParams, url, connectionTimeout, socketTimeout)
+
+        // WHEN + THEN
+        assertFailsWith<IllegalArgumentException>("The API does not use path parameters.") {
+            localUriInput.getPathParams()
+        }
+    }
+
+    @Test
+    fun `test getPathParams with path params containing illegal characters`() {
+        var testCount = 0 // Start off with count of 1 to account for ApiType.BLANK
+        ILLEGAL_PATH_PARAMETER_CHARACTERS.forEach { character ->
+            // GIVEN
+            pathParams = "index1,index2,$character,index4,index5"
+            val localUriInput = LocalUriInput(scheme, host, port, path, pathParams, queryParams, url, connectionTimeout, socketTimeout)
+
+            // WHEN + THEN
+            assertFailsWith<IllegalArgumentException>(
+                "The provided path parameters contain invalid characters or spaces. Please omit: ${ILLEGAL_PATH_PARAMETER_CHARACTERS.joinToString(" ")}"
+            ) {
+                localUriInput.getPathParams()
+            }
+            testCount++
+        }
+        assertEquals(ILLEGAL_PATH_PARAMETER_CHARACTERS.size, testCount)
+    }
+
+    @Test
+    fun `test LocalUriInput correctly determines ApiType when path is provided as URI component`() {
+        var testCount = 1 // Start off with count of 1 to account for ApiType.BLANK
+        LocalUriInput.ApiType.values()
+            .filter { enum -> enum != LocalUriInput.ApiType.BLANK }
+            .forEach { testApiType ->
+                // GIVEN
+                path = testApiType.defaultPath
+
+                // WHEN
+                val localUriInput = LocalUriInput(scheme, host, port, path, pathParams, queryParams, url, connectionTimeout, socketTimeout)
+
+                // THEN
+                assertEquals(testApiType, localUriInput.apiType)
+                testCount++
+            }
+        assertEquals(LocalUriInput.ApiType.values().size, testCount)
+    }
+
+    @Test
+    fun `test LocalUriInput correctly determines ApiType when path and path params are provided as URI components`() {
+        var testCount = 1 // Start off with count of 1 to account for ApiType.BLANK
+        LocalUriInput.ApiType.values()
+            .filter { enum -> enum != LocalUriInput.ApiType.BLANK }
+            .forEach { testApiType ->
+                // GIVEN
+                path = testApiType.defaultPath
+                pathParams = "index1,index2,index3,index4,index5"
+
+                // WHEN
+                val localUriInput = LocalUriInput(scheme, host, port, path, pathParams, queryParams, url, connectionTimeout, socketTimeout)
+
+                // THEN
+                assertEquals(testApiType, localUriInput.apiType)
+                testCount++
+            }
+        assertEquals(LocalUriInput.ApiType.values().size, testCount)
```

Comment:
NP: Up to you but you can exclude the count assertion since if all the `assertEquals` in the `forEach` pass then we should be good and if any of them fail, the test will fail anyway.